flink-yarn/src/main/java/org/apache/flink/yarn/FlinkYarnClient.java

```diff
@@ -0,0 +1,653 @@
+		FsPermission permission = new FsPermission(FsAction.ALL, FsAction.ALL, FsAction.ALL);
+		fs.setPermission(sessionFilesDir, permission); // set permission for path.
+		Utils.setTokensFor(amContainer, paths, this.conf);
+
+
+		amContainer.setLocalResources(localResources);
+		fs.close();
+
+		// Setup CLASSPATH for ApplicationMaster
+		Map<String, String> appMasterEnv = new HashMap<String, String>();
+		Utils.setupEnv(conf, appMasterEnv);
+		// set configuration values
+		appMasterEnv.put(FlinkYarnClient.ENV_TM_COUNT, String.valueOf(taskManagerCount));
+		appMasterEnv.put(FlinkYarnClient.ENV_TM_MEMORY, String.valueOf(taskManagerMemoryMb));
+		appMasterEnv.put(FlinkYarnClient.FLINK_JAR_PATH, remotePathJar.toString() );
+		appMasterEnv.put(FlinkYarnClient.ENV_APP_ID, appId.toString());
+		appMasterEnv.put(FlinkYarnClient.ENV_CLIENT_HOME_DIR, fs.getHomeDirectory().toString());
+		appMasterEnv.put(FlinkYarnClient.ENV_CLIENT_SHIP_FILES, envShipFileList.toString() );
+		appMasterEnv.put(FlinkYarnClient.ENV_CLIENT_USERNAME, UserGroupInformation.getCurrentUser().getShortUserName());
+		appMasterEnv.put(FlinkYarnClient.ENV_SLOTS, String.valueOf(slots));
+		if(dynamicPropertiesEncoded != null) {
+			appMasterEnv.put(FlinkYarnClient.ENV_DYNAMIC_PROPERTIES, dynamicPropertiesEncoded);
+		}
+
+		amContainer.setEnvironment(appMasterEnv);
+
+		// Set up resource type requirements for ApplicationMaster
+		Resource capability = Records.newRecord(Resource.class);
+		capability.setMemory(jobManagerMemoryMb);
+		capability.setVirtualCores(1);
+
+		if(clusterName == null) {
+			clusterName = "Flink session with "+taskManagerCount+" TaskManagers";
+		}
+
+		appContext.setApplicationName(clusterName); // application name
+		appContext.setApplicationType("Apache Flink");
+		appContext.setAMContainerSpec(amContainer);
+		appContext.setResource(capability);
+		appContext.setQueue(yarnQueue);
+
+
+		LOG.info("Submitting application master " + appId);
+		yarnClient.submitApplication(appContext);
+
+		LOG.info("Waiting for the cluster to be allocated");
+		int waittime = 0;
+		loop: while( true ) {
+			ApplicationReport report = yarnClient.getApplicationReport(appId);
+			YarnApplicationState appState = report.getYarnApplicationState();
+			switch(appState) {
+				case FAILED:
+				case FINISHED:
+				case KILLED:
+					throw new YarnDeploymentException("The YARN application unexpectedly switched to state "
+							+ appState +" during deployment. \n" +
+							"Diagnostics from YARN: "+report.getDiagnostics() + "\n" +
+							"If log aggregation is enabled on your cluster, use this command to further invesitage the issue:\n" +
+							"yarn logs -applicationId "+appId);
+					//break ..
+				case RUNNING:
+					LOG.info("YARN application has been deployed successfully.");
+					break loop;
+				default:
+					LOG.info("Deploying cluster, current state "+appState);
+					if(waittime > 60000) {
```

Comment:
Yeah you're right. Somehow thought that something important would depend on this threshold.